backend/src/test/java/com/google/sps/SearchMatchHelperTest.java

```diff
@@ -0,0 +1,208 @@
+import com.google.sps.data.Match;
+import com.google.sps.data.MatchPreference;
+import com.google.sps.data.MatchStatus;
+import com.google.sps.data.Participant;
+import com.google.sps.datastore.MatchDatastore;
+import com.google.sps.datastore.ParticipantDatastore;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class SearchMatchHelperTest {
+
+  // Inputs
+  private static final String USERNAME_PERSON_A = "persona";
+  private static final String USERNAME_PERSON_B = "personb";
+  private static final int DURATION_DEFAULT = 30;
+  private static final long START_TIME_AVAILABLE_DEFAULT = System.currentTimeMillis();
+  private static final long END_TIME_AVAILABLE_DURATION =
+      START_TIME_AVAILABLE_DEFAULT + TimeUnit.MINUTES.toMillis(DURATION_DEFAULT);
+  private static final long END_TIME_AVAILABLE_DEFAULT =
+      START_TIME_AVAILABLE_DEFAULT + TimeUnit.MINUTES.toMillis(100);
+  private static final String ROLE_DEFAULT = "Software engineer";
+  private static final String PRODUCT_AREA_DEFAULT = "Ads";
+  private static final List<String> INTERESTS_DEFAULT = Arrays.asList("Books");
+  private static final MatchPreference MATCH_PREFERENCE_DEFAULT = MatchPreference.ANY;
+  private static final long MATCH_ID_DEFAULT = 123456;
+  private static final MatchStatus MATCH_STATUS_MATCHED = MatchStatus.MATCHED;
+  private static final MatchStatus MATCH_STATUS_UNMATCHED = MatchStatus.UNMATCHED;
+  private static final long TIMESTAMP_DEFAULT = 0;
+
+  private HttpServletRequest request;
+  private HttpServletResponse response;
+  private MatchDatastore matchDatastore;
+  private ParticipantDatastore participantDatastore;
+  private UsernameService usernameService;
+  private SearchMatchHelper searchMatchHelper;
+
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  @Before
+  public void setUp() throws IOException {
+    request = mock(HttpServletRequest.class);
+    response = mock(HttpServletResponse.class);
+    matchDatastore = mock(MatchDatastore.class);
+    participantDatastore = mock(ParticipantDatastore.class);
+    usernameService = mock(UsernameService.class);
+
+    when(response.getWriter()).thenReturn(getWriter());
+    when(usernameService.getUsername()).thenReturn(USERNAME_PERSON_A);
+
+    helper.setUp();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void noParticipantInDatastore() throws IOException {
+    searchMatchHelper =
+        new SearchMatchHelper(matchDatastore, participantDatastore, usernameService);
```

Comment:
Since this happens in all the tests, can it move into the setUp method?

Reply:
Even though it happens in all the tests, 4/5 tests don't actually use the mock datastores since testing the different paths through the servlet requires actually putting entities into the datastore.